### PR TITLE
Attempt builds on arm64 workers

### PR DESCRIPTION
**Pull request by addyess**
Include arm64 runner to build the arm64 rock

🚨 Depends on updated workflow 🚨 
* https://github.com/canonical/k8s-workflows/pull/4

### CODE REVIEW

**Review by berkayoz — COMMENTED** [review]
Looks good overall, let's update the workflow branch reference for `k8s-workflows` to `main` (approved the k8s-workflows PR)

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -8,15 +8,24 @@ on:
 jobs:
   build-and-push-arch-specifics:
     name: Build Rocks and Push Arch Specific Images
-    uses: canonical/k8s-workflows/.github/workflows/build_rocks.yaml@main
+    uses: canonical/k8s-workflows/.github/workflows/build_rocks.yaml@KU-158/arm-builder
```
Comment (addyess) [inline .github/workflows/pull_request.yaml:11]:
```suggestion
    uses: canonical/k8s-workflows/.github/workflows/build_rocks.yaml@main
```

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -8,15 +8,24 @@ on:
...
     with:
       owner: ${{ github.repository_owner }}
       trivy-image-config: "trivy.yaml"
       multiarch-awareness: true
       cache-action: ${{ (github.event_name == 'push') && 'save' || 'restore' }}
+      platform-labels: '{"arm64": ["Ubuntu_ARM64_4C_16G_01"]}'
+  scan-images:
+    uses: canonical/k8s-workflows/.github/workflows/scan_images.yaml@KU-158/arm-builder
```
Comment (addyess) [inline .github/workflows/pull_request.yaml:19]:
```suggestion
    uses: canonical/k8s-workflows/.github/workflows/scan_images.yaml@main
```

File: .github/workflows/pull_request.yaml [hunk 3]
```diff
@@ -8,15 +8,24 @@ on:
...
+    needs: [build-and-push-arch-specifics]
+    secrets: inherit
+    with:
+      upload-result: ${{ github.event_name == 'push' }}
+      images: ${{ needs.build-and-push-arch-specifics.outputs.images }}
+      trivy-image-config: ./trivy.yaml
   build-and-push-multiarch-manifest:
     name: Combine Rocks and Push Multiarch Manifest
-    uses: canonical/k8s-workflows/.github/workflows/assemble_multiarch_image.yaml@main
+    uses: canonical/k8s-workflows/.github/workflows/assemble_multiarch_image.yaml@KU-158/arm-builder
```
Comment (addyess) [inline .github/workflows/pull_request.yaml:28]:
```suggestion
    uses: canonical/k8s-workflows/.github/workflows/assemble_multiarch_image.yaml@main
```

**Review by berkayoz — APPROVED** [review]
LGTM

**Pull request merged**